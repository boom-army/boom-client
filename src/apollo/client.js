import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
