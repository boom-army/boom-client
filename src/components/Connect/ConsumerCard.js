import React from "react";
import {
  Avatar,
  Card,
  CardActions,
  CardActionArea,
  CardContent,
  CardMedia,
  Stack,
  Typography,
} from "@mui/material";
import { Follow } from "../Profile/Follow";
import { useNavigate } from "react-router-dom";
import { styled } from "@mui/material/styles";

const CardStyled = styled(Card)`
  & .MuiButtonBase-root {
    display: block;
  }
`;

export const ConsumerCard = ({ consumer }) => {
  const navigate = useNavigate();
  const handleOnClick = () => navigate(`/${consumer.handle}`);
  return (
    <CardStyled sx={{ maxWidth: 345, height: "100%" }}>
      <CardActionArea onClick={handleOnClick}>
        <CardMedia
          component="img"
          height="80"
          image={
            consumer.coverPhoto
              ? consumer.coverPhoto
              : `${window.location.origin}/default-cover.png`
          }
          alt={`${consumer.handle} cover photo`}
        />
        <CardContent>
          <Stack direction="row" spacing={2} sx={{ marginBottom: 2 }}>
            <Avatar
              alt={`${consumer.handle} cover photo`}
              src={consumer.avatar}
            />
            <Stack>
              <Typography
                variant="body1"
                component="div"
                sx={{ marginTop: -0.7 }}
              >
                {consumer.consumerName}
              </Typography>
              <Typography variant="body2" component="div">
                @{consumer.handle}
              </Typography>
            </Stack>
          </Stack>
          <Typography variant="body2">{consumer.bio}</Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Follow sm id={consumer.id} isFollowing={consumer.isFollowing} />
      </CardActions>
    </CardStyled>
  );
};
