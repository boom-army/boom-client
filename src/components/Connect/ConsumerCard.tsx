import React from "react";
import {
  Card,
  CardActions,
  CardActionArea,
  CardContent,
  CardMedia,
  Stack,
  Typography,
} from "@mui/material";
import { Follow } from "../Profile/Follow";
import { useNavigate } from "react-router-dom";
import { styled } from "@mui/material/styles";
import { UserAvatar } from "../UserAvatar";
import { ProfileQuery } from "../../generated/graphql";

const CardStyled = styled(Card)({
  "& .MuiButtonBase-root": {
    display: "block",
  },
});
const TypographyStyled = styled(Typography)<{
  nowrap?: any;
  sx?: any;
  component?: any;
  props?: any;
}>((props) => ({}));

export const ConsumerCard = ({ profile }: ProfileQuery) => {
  const navigate = useNavigate();
  const handleOnClick = () => navigate(`/${profile.handle}`);
  return (
    <CardStyled sx={{ maxWidth: 345, height: "100%" }}>
      <CardActionArea onClick={handleOnClick}>
        <CardMedia
          component="img"
          height="80"
          image={
            profile.coverPhoto
              ? profile.coverPhoto
              : '/assets/default-cover.png'
          }
          alt={`${profile.handle} cover photo`}
        />
        <CardContent>
          <Stack direction="row" spacing={2} sx={{ marginBottom: 2 }}>
            <UserAvatar
              sx={{
                width: 30,
                height: 30,
              }}
              avatar={profile?.avatar}
              isNFT={profile?.data?.avatarIsNFT ?? false}
            />
            <Stack>
              <TypographyStyled
                variant="body1"
                component="div"
                nowrap={true && "true"}
                sx={{ marginTop: -0.7 }}
              >
                {profile.consumerName}
              </TypographyStyled>
              <TypographyStyled
                nowrap={true && "true"}
                variant="body2"
                component="div"
              >
                @{profile.handle}
              </TypographyStyled>
            </Stack>
          </Stack>
          <Typography variant="body2">{profile.bio}</Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Follow sm id={profile.id} isFollowing={profile.isFollowing} />
      </CardActions>
    </CardStyled>
  );
};
