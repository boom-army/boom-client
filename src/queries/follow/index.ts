import gql from "graphql-tag";

export const FOLLOW = gql`
  mutation follow($id: ID!) {
    follow(id: $id)
  }
`;

export const UNFOLLOW = gql`
  mutation unfollow($id: ID!) {
    unfollow(id: $id)
  }
`;

export const USERS = gql`
  query users($offset: Int, $limit: Int, $filter: String) {
    users(offset: $offset, limit: $limit, filter: $filter) {
      id
      handle
      isFollowing
      isSelf
      consumerName
      avatar
      data {
        avatarMint
        avatarUpdateAuthority
      }
      coverPhoto
      bio
    }
  }
`;

export const USER_FOLLOW = gql`
  query userFollow($offset: Int, $limit: Int) {
    userFollow(offset: $offset, limit: $limit) {
      id
      handle
      avatar
      data {
        avatarMint
        avatarUpdateAuthority
      }
      isFollowing
      isSelf
      consumerName
      bio
    }
  }
`;
