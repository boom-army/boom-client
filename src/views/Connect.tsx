import React from "react";
import { Connect } from "../components/Connect";

export const ConnectView = () => {
  return <Connect />;
};
