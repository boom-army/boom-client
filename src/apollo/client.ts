import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
