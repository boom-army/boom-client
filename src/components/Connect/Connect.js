import React, { useEffect } from "react";
import { CustomResponse } from "../CustomResponse";
import Grid from "@mui/material/Grid";
import { ConsumerCard } from "./ConsumerCard";
import { Loader } from "../Loader";
import { USERS } from "../../queries/follow";
import { useQuery } from "@apollo/client";

export const Connect = () => {
  const { loading, data, fetchMore } = useQuery(USERS, {
    variables: {
      offset: 0,
    },
  });

  useEffect(() => {
    //  TODO: This needs to be consolidated into a util
    const handleScroll = () => {
      const bottom =
        Math.ceil(window.innerHeight + window.scrollY) >=
        document.documentElement.scrollHeight;
      if (bottom) {
        fetchMore({
          variables: {
            offset: data?.users?.length,
          },
        });
      }
    };
    window.addEventListener("scroll", handleScroll, {
      passive: true,
    });
    return () => {
      window.removeEventListener("scroll", handleScroll);
    };
  }, [data, fetchMore]);

  if (loading) return <Loader />;

  return (
    <>
      {data?.users?.length ? (
        <Grid
          container
          spacing={{ md: 2 }}
          sx={{ paddingTop: 3, paddingLeft: 2, paddingRight: 2 }}
        >
          {data.users.map((consumer) => (
            <Grid item md={4} key={`grid-${consumer.id}`}>
              <ConsumerCard key={`card-${consumer.id}`} consumer={consumer} />
            </Grid>
          ))}
        </Grid>
      ) : (
        <CustomResponse text="No other Creators to follow right now" />
      )}
    </>
  );
};
