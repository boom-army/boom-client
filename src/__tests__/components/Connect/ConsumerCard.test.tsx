import React from "react";
import { ConsumerCard } from "../../../components/Connect/ConsumerCard";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { SnackbarProvider } from "notistack";

import { MockedProvider } from "@apollo/client/testing";
import { MemoryRouter } from "react-router-dom";
import { USERS } from "../../../queries/follow/index";

const consumer = {
  id: "ckzny1iv10031lmn568gj3b3n",
  handle: "shy-cloud-4965",
  avatar: "",
  isFollowing: false,
  isSelf: true,
  coverPhoto: "https://my.sosol.app/default-cover.png",
  consumerName: "shy-cloud-4965",
  bio: "full stack developer",
  __typename: "User",
};

export const __mocks__: any = [
  {
    request: {
      query: USERS,
      variables: {
        offset: 0,
      },
    },
    result: {
      data: {
        users: [
          {
            id: "ckzny1iv10031lmn568gj3b3n",
            handle: "shy-cloud-4965",
            avatar: "",
            isFollowing: false,
            isSelf: true,
            coverPhoto: "https://my.sosol.app/default-cover.png",
            consumerName: "shy-cloud-4965",
            bio: "full stack developer",
            __typename: "User",
          },
          {
            avatar: "",
            bio: "",
            consumerName: "shiny-silence-3873",
            coverPhoto: "https://my.sosol.app/default-cover.png",
            handle: "shiny-silence-3873",
            id: "cl0tpyv2k22515001mp1vbzhegs",
            isFollowing: false,
            isSelf: false,
            __typename: "User",
          },
        ],
      },
    },
  },
];

describe("<ConsumerCArd/> component :", () => {
  test("display <ConsumerCArd/> component ", async () => {
    let rendered;
    await act(async () => {
      rendered = render(
        <MockedProvider mocks={__mocks__} addTypename={false}>
          <SnackbarProvider>
            <ConsumerCard consumer={consumer} />
          </SnackbarProvider>
        </MockedProvider>,
        { wrapper: MemoryRouter }
      );
    });
    expect(rendered).toMatchSnapshot();
  });
});

test("Assertion testing of  <ConsumerCard/>  component ", async () => {
  render(
    <MockedProvider mocks={__mocks__} addTypename={false}>
      <SnackbarProvider>
        <ConsumerCard consumer={consumer} />
      </SnackbarProvider>
    </MockedProvider>,
    { wrapper: MemoryRouter }
  );
  expect(
    screen.getByRole("img", { name: "shy-cloud-4965 cover photo" })
  ).toHaveAttribute("src", "https://my.sosol.app/default-cover.png");
});
