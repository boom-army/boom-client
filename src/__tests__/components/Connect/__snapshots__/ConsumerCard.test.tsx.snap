// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsumerCArd/> component : display <ConsumerCArd/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-14 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  max-width: 345px;
  height: 100%;
}

.css-14 .MuiButtonBase-root {
  display: block;
}

.css-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: block;
  text-align: inherit;
  width: 100%;
}

.css-1::-moz-focus-inner {
  border-style: none;
}

.css-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .css-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.css-1:hover .MuiCardActionArea-focusHighlight {
  opacity: 0.04;
}

@media (hover: none) {
  .css-1:hover .MuiCardActionArea-focusHighlight {
    opacity: 0;
  }
}

.css-1.Mui-focusVisible .MuiCardActionArea-focusHighlight {
  opacity: 0.12;
}

.css-2 {
  display: block;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  -webkit-background-position: center;
  background-position: center;
  width: 100%;
  object-fit: cover;
}

.css-3 {
  padding: 16px;
}

.css-3:last-child {
  padding-bottom: 24px;
}

.css-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 16px;
}

.css-4>:not(style)+:not(style) {
  margin: 0;
  margin-left: 16px;
}

.css-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  background-color: #bdbdbd;
}

.css-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 75%;
  height: 75%;
}

.css-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.css-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  margin-top: -5.6px;
}

.css-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

.css-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

.css-11 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
  opacity: 0;
  background-color: currentcolor;
  -webkit-transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.css-12 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.css-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
}

.css-13>:not(:first-of-type) {
  margin-left: 8px;
}

.css-15 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
  cursor: pointer;
}

.css-15:hover {
  text-decoration-color: inherit;
}

.css-16 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-1"
          tabindex="0"
          type="button"
        >
          <img
            alt="shy-cloud-4965 cover photo"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-2"
            height="80"
            src="https://my.sosol.app/default-cover.png"
          />
          <div
            class="MuiCardContent-root css-3"
          >
            <div
              class="css-4"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-5"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-6"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <div
                class="css-7"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 css-8"
                  nowrap="true"
                >
                  shy-cloud-4965
                </div>
                <div
                  class="MuiTypography-root MuiTypography-body2 css-9"
                  nowrap="true"
                >
                  @
                  shy-cloud-4965
                </div>
              </div>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-10"
            >
              full stack developer
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-11"
          />
          <span
            class="MuiTouchRipple-root css-12"
          />
        </button>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-13"
        >
          <div
            class="MuiBox-root css-14"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-15"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16"
                data-testid="PersonAddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .css-14 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  max-width: 345px;
  height: 100%;
}

.css-14 .MuiButtonBase-root {
  display: block;
}

.css-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: block;
  text-align: inherit;
  width: 100%;
}

.css-1::-moz-focus-inner {
  border-style: none;
}

.css-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .css-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.css-1:hover .MuiCardActionArea-focusHighlight {
  opacity: 0.04;
}

@media (hover: none) {
  .css-1:hover .MuiCardActionArea-focusHighlight {
    opacity: 0;
  }
}

.css-1.Mui-focusVisible .MuiCardActionArea-focusHighlight {
  opacity: 0.12;
}

.css-2 {
  display: block;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  -webkit-background-position: center;
  background-position: center;
  width: 100%;
  object-fit: cover;
}

.css-3 {
  padding: 16px;
}

.css-3:last-child {
  padding-bottom: 24px;
}

.css-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 16px;
}

.css-4>:not(style)+:not(style) {
  margin: 0;
  margin-left: 16px;
}

.css-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  background-color: #bdbdbd;
}

.css-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 75%;
  height: 75%;
}

.css-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.css-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  margin-top: -5.6px;
}

.css-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

.css-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

.css-11 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
  opacity: 0;
  background-color: currentcolor;
  -webkit-transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.css-12 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.css-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
}

.css-13>:not(:first-of-type) {
  margin-left: 8px;
}

.css-15 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
  cursor: pointer;
}

.css-15:hover {
  text-decoration-color: inherit;
}

.css-16 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1"
        tabindex="0"
        type="button"
      >
        <img
          alt="shy-cloud-4965 cover photo"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-2"
          height="80"
          src="https://my.sosol.app/default-cover.png"
        />
        <div
          class="MuiCardContent-root css-3"
        >
          <div
            class="css-4"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-6"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div
              class="css-7"
            >
              <div
                class="MuiTypography-root MuiTypography-body1 css-8"
                nowrap="true"
              >
                shy-cloud-4965
              </div>
              <div
                class="MuiTypography-root MuiTypography-body2 css-9"
                nowrap="true"
              >
                @
                shy-cloud-4965
              </div>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-10"
          >
            full stack developer
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-11"
        />
        <span
          class="MuiTouchRipple-root css-12"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-13"
      >
        <div
          class="MuiBox-root css-14"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-15"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16"
              data-testid="PersonAddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
